realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -157,19 +142,20 @@ endif()
 #     -Wno-missing-field-initializers disable in object store as well.
 set(WARNING_CXX_FLAGS "-Werror -Wall -Wextra -pedantic -Wmissing-declarations \
     -Wempty-body -Wparentheses -Wunknown-pragmas -Wunreachable-code \
-    -Wno-missing-field-initializers  -Wno-maybe-uninitialized -Wno-uninitialized")
-set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_ANDROID -DREALM_HAVE_CONFIG -DPIC -pthread -fvisibility=hidden -std=c++14 -fsigned-char")
+    -Wno-missing-field-initializers")
+set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_ANDROID -DREALM_HAVE_CONFIG -DPIC -pthread -fpic -frtti -fvisibility=hidden -fvisibility-inlines-hidden -fsigned-char -std=c++14")
 if (build_SYNC)
     set(REALM_COMMON_CXX_FLAGS "${REALM_COMMON_CXX_FLAGS} -DREALM_ENABLE_SYNC=1")
 endif()
-# There might be an issue with -Os of ndk gcc 4.9. It will hang the encryption related tests.
-# And this issue doesn't seem to impact the core compiling.
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
-#-ggdb doesn't play well with -flto
+set(CMAKE_CXX_FLAGS_RELEASE "-Oz -ffunction-sections -fdata-sections -fomit-frame-pointer -g0")
```

Comment:
Why `-g0`? Will binary files be smaller if omitted?

Reply:
Actually, we don't need the -g0 flag as we strip the final binaries anyway, and I suspect the debugging information might actually be helpful in some cases.